remote-policy-xml-json-conversion/src/main/java/com/google/apigee/XmlJsonConversion.java

```diff
@@ -0,0 +1,159 @@
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import org.json.JSONObject;
+import org.json.XML;
+
+/**
+ * Decodes an Execution Protocol Buffer Message and converts Target Request Message content from XML
+ * to JSON or vice versa depending on specified flow variable. Designed to be deployed as a Google
+ * Cloud Function.
+ */
+public class XmlJsonConversion implements HttpFunction {
+
+  private final static String CONVERSION_FLOW_VARIABLE = "conversion";
+  private final static String XML_TO_JSON = "xmltojson";
+  private final static String JSON_TO_XML = "jsontoxml";
+
+  /**
+   * Reads and decodes the Execution Protocol Buffer Message and converts XML to JSON or vice versa
+   * depending on specified flow variable.
+   *
+   * @param request  contents of HTTP request containing the Protocol Buffer Message bytes
+   * @param response HttpResponse sent in response to Http request containing modified Protocol
+   *                 Buffer Message with new flow variable set
+   * @throws IOException
+   */
+  @Override
+  public void service(HttpRequest request, HttpResponse response) throws IOException {
+    try {
+      InputStream is = request.getInputStream();
+      byte[] data = is.readAllBytes();
+      Execute.Execution execution = Execute.Execution.parseFrom(data);
+      execution = convert(execution);
+      execution = setExecutionResult(execution, Action.CONTINUE, null);
+      response.getOutputStream().write(execution.toByteArray());
+    } catch (Throwable throwable) {
+      Execute.Execution execution = Execution.getDefaultInstance();
+      execution = setExecutionResult(execution, Action.ABORT, throwable);
+      response.getOutputStream().write(execution.toByteArray());
+    }
+  }
+
+  /**
+   * Delegates conversion of XML to JSON or JSON to XML depending on CONVERSION_FLOW_VARIABLE
+   * value.
+   *
+   * @param execution Execute.Execution object from caller.
+   * @return Execute.Execution object with converted content inside Target Request Message.
+   * @throws Exception
+   */
+  private Execute.Execution convert(Execute.Execution execution)
+      throws UnsupportedEncodingException {
+    validateExecution(execution);
+    String conversion = execution.getMessageContext().getTargetRequestMessage()
+        .getFlowVariablesMap().get(CONVERSION_FLOW_VARIABLE).getFlowVariable();
+    String content = execution.getMessageContext().getTargetRequestMessage().getContent()
+        .toStringUtf8();
+    if (XML_TO_JSON.equals(conversion)) {
+      execution = xmlToJson(execution, content);
+    } else if (JSON_TO_XML.equals(conversion)) {
+      execution = jsonToXml(execution, content);
+    } else {
+      throw new IllegalArgumentException("invalid conversion variable");
```

Comment:
print the what's in conversion?  That way debugging is easier on the server side.